src/main/scala/org/antipathy/scoozie/action/Hive2Action.scala

```diff
@@ -0,0 +1,145 @@
+  *    limitations under the License.
+  */
+package org.antipathy.scoozie.action
+
+import com.typesafe.config.Config
+import org.antipathy.scoozie.action.prepare.Prepare
+import org.antipathy.scoozie.builder.{ConfigurationBuilder, HoconConstants, MonadBuilder, PrepareBuilder}
+import org.antipathy.scoozie.configuration._
+import org.antipathy.scoozie.exception.ConfigurationMissingException
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable._
+import scala.xml.Elem
+
+/**
+  * Oozie Hive2 action definition
+  *
+  * @param name the name of the action
+  * @param scriptName the name of the hive script
+  * @param scriptLocation the path to the hive script
+  * @param jdbcUrl the Jdbc url to connect to Hive2 server from Beeline, required
+  * @param parameters a collection of parameters to the hive script
+  * @param jobXmlOption optional job.xml path
+  * @param files additional files to pass to job
+  * @param configuration additional config for this action
+  * @param yarnConfig Yarn configuration for this action
+  * @param prepareOption an optional prepare stage for the action
+  */
+final class Hive2Action(override val name: String,
+                        scriptName: String,
+                        scriptLocation: String,
+                        jdbcUrl: String,
+                        parameters: Seq[String],
+                        override val jobXmlOption: Option[String],
+                        files: Seq[String],
+                        override val configuration: Configuration,
+                        yarnConfig: YarnConfig,
+                        override val prepareOption: Option[Prepare])
+    extends Action
+    with HasPrepare
+    with HasConfig
+    with HasJobXml {
+
+  private val scriptNameProperty = formatProperty(s"${name}_scriptName")
+  private val scriptLocationProperty = formatProperty(s"${name}_scriptLocation")
+  private val jdbcUrlProperty = formatProperty(s"${name}_jdbcUrl")
+  private val parametersProperties =
+    buildSequenceProperties(name, "parameter", parameters)
+  private val filesProperties = buildSequenceProperties(name, "file", files)
+
+  /**
+    * Get the Oozie properties for this object
+    */
+  override def properties: Map[String, String] =
+    jobXmlProperty ++
+    yarnConfig.properties ++
+    Map(scriptNameProperty -> scriptName, scriptLocationProperty -> scriptLocation, jdbcUrlProperty -> jdbcUrl) ++
+    prepareProperties ++ parametersProperties ++ mappedProperties ++ filesProperties
+
+  /**
+    * The XML namespace for an action element
+    */
+  override val xmlns: Option[String] = Some("uri:oozie:hive2-action:0.1")
+
+  /**
+    * The XML for this node
+    */
+  override def toXML: Elem =
+    <hive2 xmlns={xmlns.orNull}>
+        {yarnConfig.jobTrackerXML}
+        {yarnConfig.nameNodeXML}
+        {prepareXML}
+        {jobXml}
+        {configXML}
+        <jdbc-url>{jdbcUrlProperty}</jdbc-url>
+        <script>{scriptNameProperty}</script>
```

Comment:
I need to fix the <script></script> logic so it has the value `path/script` instead of just the script here